applications/clock_app/clock_app.c

```diff
@@ -0,0 +1,412 @@
+#include <furi.h>
+#include <furi_hal.h>
+#include <gui/elements.h>
+#include <gui/gui.h>
+#include <input/input.h>
+#include <notification/notification.h>
+#include <notification/notification_messages.h>
+
+#define TAG "Clock"
+
+bool timerStarted = false;
+int timerSecs = 0;
+int songSelect = 2;
+
+typedef enum {
+    EventTypeTick,
+    EventTypeKey,
+} EventType;
+
+typedef struct {
+    EventType type;
+    InputEvent input;
+} PluginEvent;
+
+typedef struct {
+    FuriHalRtcDateTime datetime;
+} ClockState;
+
+static void clock_input_callback(InputEvent* input_event, FuriMessageQueue* event_queue) {
+    furi_assert(event_queue);
+    PluginEvent event = {.type = EventTypeKey, .input = *input_event};
+    furi_message_queue_put(event_queue, &event, FuriWaitForever);
+}
+
+static void clock_render_callback(Canvas* const canvas, void* ctx) {
+    canvas_clear(canvas);
+    canvas_set_color(canvas, ColorBlack);
+    ClockState* state = (ClockState*)acquire_mutex((ValueMutex*)ctx, 25);
+    char strings[3][20];
+    int curMin = (timerSecs / 60);
+    int curSec = timerSecs - (curMin * 60);
+    snprintf(
+        strings[0],
+        20,
+        "%.4d-%.2d-%.2d",
+        state->datetime.year,
+        state->datetime.month,
+        state->datetime.day);
+    snprintf(
+        strings[1],
+        20,
+        "%.2d:%.2d:%.2d",
+        state->datetime.hour,
+        state->datetime.minute,
+        state->datetime.second);
+    snprintf(strings[2], 20, "%.2d:%.2d", curMin, curSec);
+    release_mutex((ValueMutex*)ctx, state);
+    canvas_set_font(canvas, FontBigNumbers);
+    if(timerStarted) canvas_draw_str_aligned(canvas, 64, 8, AlignCenter, AlignCenter, strings[1]);
+    if(!timerStarted)
+        canvas_draw_str_aligned(canvas, 64, 26, AlignCenter, AlignCenter, strings[1]);
+    canvas_set_font(canvas, FontSecondary);
+    if(timerStarted) canvas_draw_str_aligned(canvas, 64, 20, AlignCenter, AlignTop, strings[0]);
+    if(!timerStarted) canvas_draw_str_aligned(canvas, 64, 38, AlignCenter, AlignTop, strings[0]);
+    // elements_button_left(canvas, "Alarms");
```

Comment:
please remove commented code

Reply:
will advise when all the changes are complete

Reply:
Removed